test/k8sT/Services.go

```diff
@@ -432,6 +432,69 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 	})
 
+	SkipContextIf(helpers.RunsWithKubeProxy, "Checks connectivity when skipping socket lb in pod ns", func() {
+		var (
+			demoDSYAML string
+			demoYAML   string
+		)
+
+		BeforeAll(func() {
+			DeployCiliumOptionsAndDNS(kubectl, ciliumFilename, map[string]string{
+				"skipSocketLB": "true",
+			})
+			demoDSYAML = helpers.ManifestGet(kubectl.BasePath(), "demo_ds.yaml")
+			res := kubectl.ApplyDefault(demoDSYAML)
+			res.ExpectSuccess("Unable to apply %s", demoDSYAML)
+			demoYAML = helpers.ManifestGet(kubectl.BasePath(), "demo.yaml")
+			res = kubectl.ApplyDefault(demoYAML)
+			res.ExpectSuccess("unable to apply %s", demoYAML)
+			waitPodsDs()
+			err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(demoDSYAML)
+			_ = kubectl.Delete(demoYAML)
+			ExpectAllPodsTerminated(kubectl)
+		})
+
+		It("Checks ClusterIP connectivity on the same node", func() {
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, serviceName)
+			Expect(err).Should(BeNil(), "Cannot get service %s", serviceName)
+			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
+
+			httpSVCURL := fmt.Sprintf("http://%s/", clusterIP)
+			tftpSVCURL := fmt.Sprintf("tftp://%s/hello", clusterIP)
+
+			// Test connectivbity from root ns
+			status := kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
+				helpers.CurlFail(httpSVCURL))
+			status.ExpectSuccess("cannot curl to service IP from host")
+			status = kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
+				helpers.CurlFail(tftpSVCURL))
+			status.ExpectSuccess("cannot curl to service IP from host")
+
+			// Test connectivity from pod ns
+			testCurlFromPods("id=app2", httpSVCURL, 10, 0)
```

Comment:
Should we check existence of the cluster IP in the `cilium monitor` output here when we have the config option enabled, and service translation doesn't happen via BPF socket prog?

Reply:
Good point, done.

Reply:
Changed how the test works, turns out `cilium monitor` won't capture cluster IP in either case, now it's checking in the ct table, we should see service VIP in ct only when tc service resolution takes effect.

Reply:
Umm... It should dump the cluster IP when socket lb is not in play. Can you try it outside the test environment?

Reply:
Discussed offline - @Weil0ng is going to add a check for the cluster IP in the monitor output in order to validate that the flag works.

Reply:
Done.